src/lib/JoinRoom.svelte

```diff
@@ -0,0 +1,14 @@
+<script lang="ts">
+	export let roomCodePattern: RegExp = /.*/; // default to match anything
```

Comment:
Looks like there's maybe some duplication of the room code pattern matching between this file and the other one, could they be unified?

Reply:
I think both are needed- The "export .." declaration inside the component basically defines what can be passed in as props. Then the declaraction in the "parent" component defines the variable they want to pass in